dipy/denoise/shift_twist_convolution.pyx

```diff
@@ -155,7 +155,8 @@ cdef double [:, :, :, ::1] perform_convolution (double [:, :, :, ::1] odfs,
         cnp.npy_intp nz = odfs.shape[2]
         cnp.npy_intp threads_to_use = -1
         cnp.npy_intp all_cores = openmp.omp_get_num_procs()
-        cnp.npy_intp corient, orient, cx, cy, cz, x, y, z
+        cnp.npy_intp corient
+        int orient = 0, cx = 0, cy = 0, cz = 0, x = 0, y = 0, z = 0
```

Comment:
Why are you changing the type here? Can you explain to me?

Reply:
These are used in regular `range`-based loops, so I thought the change was fair. But again, they keep throwing warnings, so depending on your investigations, these may need to be reverses or changed.

Reply:
I would revert this and also the other one that @skoudoro flagged.

Reply:
Thanks for the review @arokem. Reverted in 10032f0.